opensky/python/publisher.py

```diff
@@ -0,0 +1,119 @@
+        with open(opensky_creds) as f:
+            self.opensky_creds = json.load(f)
+            if (
+                "username" not in self.opensky_creds
+                or "password" not in self.opensky_creds
+            ):
+                raise ValueError(
+                    "Opensky credentials must contain both username and password"
+                )
+
+    def run(self):
+        """
+        Run the publisher forever.
+        """
+        asyncio.get_event_loop().run_until_complete(self.recv_and_publish())
+
+    async def print_ack(self, ack):
+        ts = datetime.fromtimestamp(ack.committed.seconds + ack.committed.nanos / 1e9)
+        print(f"Event committed at {ts}")
+
+    async def print_nack(nack):
+        print(f"Event was not committed with error {nack.code}: {nack.error}")
+
+    async def recv_and_publish(self):
+        """
+        Retrieve flight data from OpenSky and publish events to Ensign.
+        """
+        await self.ensign.ensure_topic_exists(self.topic)
+
+        async with ClientSession() as session:
+            async with OpenSky(session=session) as opensky:
+                opensky.authenticate(
+                    BasicAuth(
+                        login=self.opensky_creds["username"],
+                        password=self.opensky_creds["password"],
+                    )
+                )
+
+                while True:
+                    try:
+                        states = await opensky.get_states(
+                            bounding_box=self.bounding_box
+                        )
+                    except Exception as e:
+                        print(e)
+                        await asyncio.sleep(self.interval)
+                        continue
+                    for event in self.states_to_events(states):
+                        await self.ensign.publish(
+                            self.topic,
+                            event,
+                            on_ack=self.print_ack,
+                            on_nack=self.print_nack,
+                        )
+
+                    await asyncio.sleep(self.interval)
+
+    def states_to_events(self, states):
+        for vector in states.states:
+            data = {
+                "icao24": vector.icao24,
+                "callsign": vector.callsign,
+                "origin_country": vector.origin_country,
+                "time_position": vector.time_position,
+                "last_contact": vector.last_contact,
+                "longitude": vector.longitude,
+                "latitude": vector.latitude,
+                "geo_altitude": vector.geo_altitude,
+                "on_ground": vector.on_ground,
+                "velocity": vector.velocity,
+                "true_track": vector.true_track,
+                "vertical_rate": vector.vertical_rate,
+                "sensors": vector.sensors,
+                "barometric_altitude": vector.barometric_altitude,
```

Comment:
In the return types [here](https://openskynetwork.github.io/opensky-api/python.html#retrieving-data), I see `baro_altitude`. Should this be updated?

Reply:
I'm using [this one](https://github.com/joostlek/python-opensky) which is not the official Python client but it's calling the same API. It was easier to get going because it's pip installable and doesn't require cloning a repo. However now that you pointed out that some of the fields do not match the docs I'm leaning towards using the official one instead.

Reply:
I saw the package mentioned in the code, but didn't realize that they could've made changes to the return type fields. I now see where they use an alias in the [models.py file](https://github.com/joostlek/python-opensky/blob/main/src/python_opensky/models.py). 

If it's easier to use the package, then I think a note should be added to the README to use what's listed in the models.py file along with a link.